src/DDoSX/CdnClient.php

```diff
@@ -30,7 +37,52 @@ public function create(Cdn $cdn)
         return (new SelfResponse($body, "domain_name"))
             ->setClient($this)
             ->serializeWith(function ($response) {
-                return new Cdn($this->apiToFriendly($response->data, $this->requestMap));
+                return new Cdn($this->apiToFriendly($response->data, self::CDN_MAP));
             });
     }
+
+    /**
+     * Get a paginated list of cdn rules for a domain
+     * @param $domainName
+     * @param int $page
+     * @param int $perPage
+     * @param array $filters
+     * @return int|\UKFast\SDK\Page
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getRules($domainName, $page = 1, $perPage = 15, $filters = [])
+    {
+        $page = $this->paginatedRequest(
+            'v1/domains/' . $domainName . '/cdn/rules',
+            $page,
+            $perPage,
+            $filters
+        );
+
+        $page->serializeWith(function ($item) {
+            return $this->serializeCdnRule($item);
+        });
+
+        return $page;
+    }
+
+    /**
+     * Get a rule for a domain by it's ID
+     * @param $domainName
+     * @param $ruleId
+     * @return CdnRule
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getRuleById($domainName, $ruleId)
+    {
+        $response = $this->get('v1/domains/' . $domainName . '/cdn/rules/' . $ruleId);
+        $body = $this->decodeJson($response->getBody()->getContents());
+
+        return $this->serializeCdnRule($body->data);
+    }
+
+    protected function serializeCdnRule($raw)
+    {
+        return new CdnRule($this->apiToFriendly($raw));
```

Comment:
```suggestion
        return new CdnRule($this->apiToFriendly($raw, self::RULES_MAP));
```

Reply:
Fixes `PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function UKFast\SDK\Client::apiToFriendly(), 1 passed`